lib/ecto/datetime.ex

```diff
@@ -214,12 +215,17 @@ defmodule Ecto.Time do
   end
 
   defp erl_load({_, {hour, min, sec}}) do
-    %Ecto.Time{hour: hour, min: min, sec: sec}
+    %Ecto.Time{hour: hour, min: min, sec: sec, usec: 0}
   end
 
   defimpl String.Chars do
-    def to_string(%Ecto.Time{hour: hour, min: min, sec: sec}) do
-      zero_pad(hour, 2) <> ":" <> zero_pad(min, 2) <> ":" <> zero_pad(sec, 2)
+    def to_string(%Ecto.Time{hour: hour, min: min, sec: sec, usec: usec}) do
+      str = zero_pad(hour, 2) <> ":" <> zero_pad(min, 2) <> ":" <> zero_pad(sec, 2)
+      if is_nil(usec) or usec == 0 do
```

Comment:
I checked the [ISO 8601 page](http://en.wikipedia.org/wiki/ISO_8601) on Wikipedia before adding that. Quoting:

> Decimal fractions may be added to any of the three time elements. However, a fraction may only be added to the lowest order time element in the representation. A decimal mark, either a comma or a dot (without any preference as stated in resolution 10 of the 22nd General Conference CGPM in 2003, but with a preference for a comma according to ISO 8601:2004) is used as a separator between the time element and its fraction. To denote "14 hours, 30 and one half minutes", do not include a seconds figure. Represent it as "14:30,5", "1430,5", "14:30.5", or "1430.5". There is no limit on the number of decimal places for the decimal fraction. However, the number of decimal places needs to be agreed to by the communicating parties.

So it *is* allowed. Nonetheless, if that's for humans it could be left out :) I implemented so that it adds microseconds to the representation only if `:usec` is not `0`. What do you think?